tuf/api/metadata.py

```diff
@@ -1026,6 +1026,10 @@ def __init__(
         unrecognized_fields: Optional[Mapping[str, Any]] = None,
     ):
         super().__init__(version, spec_version, expires, unrecognized_fields)
+        if any(meta_path == "" for meta_path in meta):
+            raise ValueError("All meta paths must be non-empty strings")
```

Comment:
you originally suggested checking that "meta_path is a non-empty string": now you check that "metapath is not an empty string". These are subtly different as the former includes type checking and the implemented check is so specific that I don't think it's worth two lines of code.

I'm still not sure how much type checking we actually should do... maybe we should validate the dictionary key types as that doesn't happen "implicitly" like it tends to happen for other data that is wrongly typed. If we don't do that we could deserialize (and then later serialize) data that is not actually  valid metadata